src/Service/ITenantProvider.cs

```diff
@@ -5,5 +5,5 @@ namespace Passwordless.Service;
 
 public interface ITenantProvider
 {
-    public string Tenant { get; set; }
+    public string Tenant { get; }
```

Comment:
Neat

Reply:
The `ManualTenantProvider` is using the constructor, which is used for the EfTenantStorageFactory that remains. Because the TenantProvider reads out the claims in the getter.